File: server/cmd/domain/Recipe.go
package domain

// Recipe struct
type Recipe struct {
	Base
	Name              string              `json:"name"`
	Tags              []*Tag              `json:"tags" gorm:"many2many:recipe_tags"`
	RecipeIngredients []*RecipeIngredient `json:"recipeIngredients"`
	PrepTime          int                 `json:"prepTime"`
	BakeTime          int                 `json:"bakeTime"`
	BakeTemp          int                 `json:"bakeTemp"`
}

File: frontend/src/domain/Recipe.ts
import { RecipeIngredient } from '.'

export default interface Recipe {
    bakeTemp: number
    bakeTime: number
    recipeIngredients: RecipeIngredient[]
    createdAt: string
    deletedAt: string | null
    id: string
    prepTime: number
    tags: string[]
    updatedAt: string
    name: string
}